Automattic/wp-calypso
# Quick Start: Prevent blocked users from scheduling an appointment

#### Changes proposed in this Pull Request

Check pcgXQF-eV-P2 and 119-gh-concierge-admin.

In this PR:
* Users blocked/banned from Happychat are prevented from purchasing a new Quick Start product.
* Quick Start upsell messages in /me/purchases and /help are not shown for blocked users.
* Users blocked/banned from Happychat with an existing Quick Start product will not be allowed to schedule an appointment.

#### Testing instructions



**Scenario 1 - Prevent blocked users from scheduling an appointment**

1. Apply D66745-code.
2. Login to an account that has an existing Quick Start product. 
3. Modify backend code line by following the instruction given in the "Testing Instructions" section of D66745-code, to force your logged in account to be blocked in Happychat.
4. Now head to `/me/quickstart` and verify that you are not able to schedule an appointment. You should see a screen like this:

<img width="752" alt="Screenshot 2021-09-16 at 5 19 04 PM" src="https://user-images.githubusercontent.com/1269602/133619542-b3df8644-b3e5-4def-97e1-5a5fa052fed0.png">


**Scenario 2 - Prevent blocked users from purchasing Quick Start**

1. Apply D66745-code and modify the backend code line by following the instruction given in the "Testing Instructions" section of D66745-code
2. Sign in to an account which does not have a Quick Start product. 
3. Navigate to `/checkout/offer-quickstart-session` and attempt to purchase.
4. You should see an error message as shown below:

<img width="1301" alt="Screenshot 2021-09-16 at 5 26 15 PM" src="https://user-images.githubusercontent.com/1269602/133620556-2ac04f07-a70b-4e1e-bc80-4c64d91ce461.png">


**Scenario 3 - Upsell banners to not be shown in Calypso**

1. Apply D66745-code and modify the backend code line by following the instruction given in the "Testing Instructions" section of D66745-code
2. Visit `/me/purchases` and `/me/help` and verify that you don't see any upsell banners for Quick Start.




Related to 119-gh-concierge-admin
